Controller/SecurityFOSUser1Controller.php

```diff
@@ -37,6 +37,41 @@ public function loginAction()
             return new RedirectResponse($url);
         }
 
-        return parent::loginAction();
+        /*
+            Implementation of parent::loginAction
+            Needed for fixing the "Bad Credentails" translation problem
+            The code is a mix of current dev-master
+            and the 1.3.6 version of FOSUserBundle
+        */
+
+        $request = $this->container->get('request');
+        /* @var $request \Symfony\Component\HttpFoundation\Request */
+        $session = $request->getSession();
+        /* @var $session \Symfony\Component\HttpFoundation\Session\Session */
+
+        // get the error if any (works with forward and redirect -- see below)
+        if ($request->attributes->has(SecurityContextInterface::AUTHENTICATION_ERROR)) {
+            $error = $request->attributes->get(SecurityContextInterface::AUTHENTICATION_ERROR);
+        } elseif (null !== $session && $session->has(SecurityContextInterface::AUTHENTICATION_ERROR)) {
+            $error = $session->get(SecurityContextInterface::AUTHENTICATION_ERROR);
+            $session->remove(SecurityContextInterface::AUTHENTICATION_ERROR);
+        } else {
+            $error = null;
+        }
+
+        if (!$error instanceof AuthenticationException) {
+            $error = null; // The value does not come from the security component.
+        }
+
+        // last username entered by the user
+        $lastUsername = (null === $session) ? '' : $session->get(SecurityContextInterface::LAST_USERNAME);
+
+        $csrfToken = $this->container->get('form.csrf_provider')->generateCsrfToken('authenticate');
```

Comment:
Both these variables are used only once, as values for keys that have the same name. You could write directly :

```php
array(
    'last_username' => (null === $session) ? '' : $session->get(SecurityContextInterface::LAST_USERNAME);
    'csrf_token' => $this->container->get('form.csrf_provider')->generateCsrfToken('authenticate');
)
```